controllers/helpers.go

```diff
@@ -149,6 +149,7 @@ func (r *EtcdadmClusterReconciler) generateMachine(ctx context.Context, ec *etcd
 				ConfigRef: bootstrapRef,
 			},
 			FailureDomain: failureDomain,
+			Version:       &ec.Spec.EtcdadmConfigSpec.CloudInitConfig.Version,
```

Comment:
I think this will panic for Bottlerocket since CloudInitConfig is a pointer.
And we also need to set the version for Bottlerocket, by parsing BottlerocketConfig.EtcdImage field